src/S3/MultipartCopy.php

```diff
@@ -128,15 +129,17 @@ private function createPart($partNumber, $partsCount)
             $data[$k] = $v;
         }
 
-        list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
-        $data['CopySource'] = '/' . $bucket . '/' . implode(
-            '/',
-            array_map(
-                'urlencode',
-                explode('/', rawurldecode($key))
-            )
-        );
+        if (is_array($this->source)) {
```

Comment:
These if/else pieces need comments to explain what behavior change is happening.